docs/troubleshooting.md

```diff
@@ -60,3 +60,44 @@ As part of [CI](../scripts/ci-e2e.sh) there is a [log collection script](hack/..
 ```bash
 ./hack/log/log-dump.sh
 ```
+## Examples of troubleshooting real-world issues
+
+### Nodes did not come online
+
+If as a result of a new cluster create operation, or as a result of adding a new machine or machinepool resource to an existing cluster, one or more nodes did not join the cluster, you can use some of the above guidance to SSH into the VM(s) and debug what happened. First, let's find out which VMs were created but failed to join the cluster by introspecting all VMs in the cluster resource group, and comparing them to the nodes present in the cluster:
+
+```
+$ export CLUSTER_RESOURCE_GROUP=my-cluster-rg
+$ export VM_PREFIX=my-cluster-md-0-
+$ export KUBECONFIG=/Users/me/.kube/my-cluster.kubeconfig
+$ $ for vm in $(az vm list -g $CLUSTER_RESOURCE_GROUP |  jq -r --arg VM_PREFIX "${VM_PREFIX}" '.[] | select (.name | startswith($VM_PREFIX)).name'); do kubectl get node $vm 2>&1 >/dev/null && continue || echo node $vm did not join the cluster; done
```

Comment:
this only works for VMs, not vmss (ie. not MachinePool) right?

Reply:
Right. I'll add an equivalent example for VMSS.

Reply:
wouldn't `kubectl get machines` / `kubectl get machinepool`  `grep -v Ready` have the same effect without needing az cli and calling to Azure? In your case did the machine show as Ready?

Reply:
Sorry, `grep -v Running` *

Reply:
I honestly didn't look at the `machine` resource at all. This was how my brain worked:

1) create cluster w/ desired node count
2) wait for nodes to come online, after a while noticed that there was one missing node
3) how many actual VMs are in my resource group? it's 20
4) O.K., so which one didn't register as a node?

FWIW

Reply:
Right, I think both are valid, you went to the RG as a first instinct because you're familiar with Azure. For users who aren't as comfortable with Azure specific stuff it'd be nice to document how all this stuff can be done without needing to care about the underlying infrastructure... `kubectl get azuremachine` should show you the VM status while `kubectl get machine` should show you the status of the machine from a k8s perspective.

Reply:
I think both perspectives are valid here.  In reality will probably need to look at both CAPZ CRD and the underlying infra. Atleast that's how I have approached it so far, I look at the crd's for a quick understanding of what CAPZ thinks is going on then I look at the azure system to see what is happening.  It would be worth add a small section to see how this vms in azure related to the CRD's of capz.